app.js

```diff
@@ -0,0 +1,78 @@
+require('express-async-errors');
+require('dotenv').config();
+
+const session = require('express-session');
+const express = require('express');
+const app = express();
+
+const MongoDBStore = require('connect-mongodb-session')(session);
+const url = process.env.MONGO_URI;
+
+const store = new MongoDBStore({
+    uri: url,
+    collection: 'sessions',
+});
+store.on('error', function (error) {
+    console.log(error);
+});
+
+const sessionParams = {
+    secret: process.env.SESSION_SECRET,
+    resave: true,
+    saveUninitialized: true,
+    store: store,
+    cookie: { secure: false, sameSite: "strict" },
+};
+
+if (app.get('env') === 'production') {
+    app.set('trust proxy', 1); // trust first proxy
+    sessionParams.cookie.secure = true; // serve secure cookies
+}
+
+app.use(session(sessionParams));
+app.use(require("connect-flash")());
+app.set('view engine', 'ejs');
+app.use(require('body-parser').urlencoded({ extended: true }));
+
+
+app.get('/secretWord', (req, res) => {
+    if (!req.session.secretWord) {
+        req.session.secretWord = "panda";
+    }
+    res.locals.info = req.flash("info");
+    res.locals.error = req.flash("error");
+    res.render("secretWord", { secretWord: req.session.secretWord });
+})
+app.post("/secretWord", (req, res) => {
+    if (req.body.secretWord.toUpperCase() == 'P') {
```

Comment:
This condition is only true if the secretword is "p" or "P" .  
secret words that start with p like pasta are allowed.

Review lesson on the code that checks if first letter is "P"

Reply:
@olearyma Fixed.. Please check